테스트코드 작성

garamv20 opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제
- member, wasteLike, wasteReview관련 테스트 코드 작성이 안되있었어서, 테스트코드 작성하고자 합니다.

## ✨ 이 PR에서 핵심적으로 변경된 사항
> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요
- member, wasteLike, wasteReview관련 테스트 코드 작성

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분
> 없으면 "없음" 이라고 기재해 주세요
- WasteLikeApi의 addOrDeleteWasteLike메소드에서 중복 코드가 있어서 수정하였습니다. 

### 📌 PR 진행 시 이러한 점들을 참고해 주세요
* Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
* Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
* Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 1일 이내에 진행해 주세요.

## Issue Tags
- Closed : #89

garamv20 committed:
refactor: 관심추가,삭제 부분 if중복값 수정

garamv20 committed:
test: #89 - 테스트코드 작성(member, wasteLike, wasteReview)

JadeKim042386 commented on src/main/java/freshtrash/freshtrashbackend/controller/WasteLikeApi.java:
```diff
@@ -44,9 +44,9 @@ public ResponseEntity<ApiResponse<Boolean>> addOrDeleteWasteLike(
         checkIfNotWriter(memberPrincipal, wasteId);
         Boolean isLike = likeStatus == LikeStatus.LIKE;
-        if (likeStatus == LikeStatus.LIKE) {
+        if (isLike) {
```
다시보니 addWasteLike와 deleteWasteLike 메소드의 중복 코드를 최소화하면 좋을 것 같습니다. 이후 리팩토링해주시는 것이 어떨까요?

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/controller/MemberApiTest.java:
```diff
@@ -0,0 +1,123 @@
+        MockMultipartFile imgFile = Fixture.createMultipartFile("test_image");
+        MemberRequest memberRequest = new MemberRequest(
+                "user111",
+                Address.builder()
```
address도 Fixture에 추가되어있습니다. 확인바랍니다.

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/controller/MemberApiTest.java:
```diff
@@ -0,0 +1,123 @@
+        given(memberService.updateMember(
+                        any(MemberPrincipal.class), any(MemberRequest.class), any(MultipartFile.class)))
+                .willReturn(member);
+        willDoNothing().given(localFileService).deleteFileIfExists(anyString());
```
deleteFileIfExists()에 anyString()을 사용하신 이유가 있을까요?

garamv20 replied:
테스트하다 오류가 나서 이것저것해보다 들어갔는데 수정이 안됐네요. eq(oldFile)로 수정할게요~!

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/controller/WasteLikeApiTest.java:
```diff
@@ -52,4 +55,34 @@ void given_loginUserAndPageable_when_getLikedWastes_then_returnPagingWasteData()
         // then
     }
+
+    @Test
```
`@ParameterizedTest`를 사용할 수 있지 않을까요?

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/service/MemberServiceTest.java:
```diff
@@ -0,0 +1,78 @@
+
+        MockMultipartFile imgFile = Fixture.createMultipartFile("test_image");
+        MemberRequest memberRequest = new MemberRequest("user111", Fixture.createAddress());
+        MemberPrincipal updatedMemberPrincipal = MemberPrincipal.builder()
```
MemberPrincipal도 FixtureDto에 추가되어있습니다. 확인바랍니다.

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/service/MemberServiceTest.java:
```diff
@@ -0,0 +1,78 @@
+        MemberPrincipal memberPrincipal = FixtureDto.createMemberPrincipal();
+
+        MockMultipartFile imgFile = Fixture.createMultipartFile("test_image");
+        MemberRequest memberRequest = new MemberRequest("user111", Fixture.createAddress());
```
MemberRequest도 FixtureDto에 추가해둔 것처럼 일관성있게 static 메소드로 정의하는게 어떨까요?

JadeKim042386 reviewed (commented):
수고하셨습니다. 전체 리뷰확인바랍니다.

garamv20 committed:
test: #89 - 테스트코드 수정
- WasteLikeApiTest @ParameterizedTest 적용
- static 메소드 활용

garamv20 commented:
@JadeKim042386 리뷰 감사합니다. 리뷰주신사항들 반영해서 수정했습니다 확인부탁드려요~

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/controller/MemberApiTest.java:
```diff
@@ -0,0 +1,115 @@
+        Long memberId = 123L;
+        String oldFile = "oldFile.png";
+        MockMultipartFile imgFile = Fixture.createMultipartFile("test_image");
+        MemberRequest memberRequest = new MemberRequest("user111", Fixture.createAddress());
```
@garamv20 MemberServiceTest에서는 FixtureDto로 MemberRequest를 생성하였는데 여기는 new로 생성하셨네요. 일관성있게 작성해주시면 좋을 것 같습니다.

garamv20 replied:
@garamv20 네 제가 놓쳤네요 수정했습니다!

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/controller/MemberApiTest.java:
```diff
@@ -0,0 +1,115 @@
+        // when
+        mvc.perform(multipart(HttpMethod.PUT, "/api/v1/members")
+                        .file("imgFile", imgFile.getBytes())
+                        .file(new MockMultipartFile(
```
@garamv20 File도 마찬가지입니다. Fixture에 이미 구현되어있는 static 메소드가 있습니다. 리뷰한 부분만이 아니라 전체 확인해주시면 좋을 것 같습니다.

garamv20 replied:
@JadeKim042386 이미 구현되있는 메소드는 이미지파일넣는거고 저는 MemberRequest 전달하는 부분이라 new로 했는데 일관성있게 static 메소드로 수정하겠습니다!

JadeKim042386 reviewed (commented):
수고하셨습니다. 리뷰 확인해주시고 한 번만 더 정리해주시면 감사하겠습니다.

garamv20 committed:
test: #89 - 테스트코드 수정
- 객체생성 static 메소드 형식으로 수정

JadeKim042386 commented on src/test/java/freshtrash/freshtrashbackend/Fixture/Fixture.java:
```diff
@@ -153,4 +157,10 @@ public static TransactionLog createTransactionLog() {
         return transactionLog;
     }
+
+    public static MockMultipartFile createMultipartFileWithName(String fileName, String content) {
```
@garamv20 인자로 fileName을 받는데 MockMutlipartFile을 생성할 때는 filename이 아니라 name에 지정하신 것 같습니다. 이유가 있을까요? 그리고 비슷한 메소드(MockMutlipartFile를 생성하는 메소드와 같은)들은 떨어져있는게 아니라 주변에 있어야지 보기 편하지 않을까요?

JadeKim042386 replied:
@garamv20 작업에 집중하기 힘드신 것 같아 해당 사항은 제가 반영했습니다.

garamv20 replied:
@JadeKim042386 네 확인이 너무 늦어 죄송해요 감사합니다..!

JadeKim042386 committed:
test: #89 - Fixture 클래스 수정
- 이미지 파일을 생성하는 메소드와 json 형식의 파일을 생성하는 메소드를 구분
- json 파일 생성 시 fileName이 인자로 넘겨지는 것이 아닌 name이 넘겨지도록 수정
